src/colorizer/Dataset.ts

```diff
@@ -122,10 +126,11 @@ export default class Dataset {
 
   /**
    * Loads a feature from the dataset, fetching its data from the provided url.
-   * @returns A promise of an array tuple containing the feature name and its FeatureData.
+   * @returns A promise of an array tuple containing the feature key and its FeatureData.
    */
   private async loadFeature(metadata: ManifestFile["features"][number]): Promise<[string, FeatureData]> {
     const name = metadata.name;
+    const key = metadata.key || getKeyFromName(name);
```

Comment:
both here and in update manifest you're calling this getKeyFromName. I don't have a full picture of the code path, but it seems like there should only be one place where you check if the key has been provided in the data and update it if not

Reply:
That makes sense! I removed the usage in `dataset_utils`.